Rewrite install task

tianshengsui opened the pull request:
resolve #1909

igboyes commented on virtool/hmm/db.py:
```diff
@@ -38,6 +38,125 @@
 ]
 
 
+class HmmInstallTask(virtool.tasks.task.Task):
```
Rename to `HMMInstallTask`.

igboyes reviewed (changes requested):
One rename.

Why didn't you split the task functions into separate steps when moving to `Task`?

tianshengsui commented:
> One rename.
> 
> Why didn't you split the task functions into separate steps when moving to `Task`?

Because all the code are under the context of ```tempdir```. If I split them into several functions, I think the ```tempdir``` will be closed and removed.

igboyes commented:
Okay. That is what I thought.

Can you add the tempdir generation and cleanup to the base `Task` class and make `temp_dir` an attribute on that class? That way the temporary directory is available to all the tasks but is still automatically deleted at the end.

tianshengsui commented:
I will do it.
